provisioning/miniprov/hare_mp/utils.py

```diff
@@ -349,6 +349,15 @@ def save_ssl_config(self):
         })
         self.kv.kv_put('ssl/hax', ssl_hax)
 
+    @func_log(func_enter, func_leave)
+    @repeat_if_fails()
+    def get_node_group(self, machine_id: str):
+        key = f'conf/node>{machine_id}>node_group'
+        node_group = self.kv.kv_get(key, allow_null=True)
```

Comment:
Where are we saving this?

Reply:
I don't think we need to use `@repeat_if_fails` if we are allowing the corresponding kv to be NULL.

Reply:
> Where are we saving this?
It will be saved by provisioner, I have comment regarding this.

Reply:
> I don't think we need to use `@repeat_if_fails` if we are allowing the corresponding kv to be NULL.
I have modified code and not allowing it to be null now